src/devices/Mpu9250/Mpu6500.cs

```diff
@@ -0,0 +1,899 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Magnometer;
+using Iot.Units;
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Device;
+using System.Device.I2c;
+using System.IO;
+using System.Numerics;
+using System.Text;
+using System.Threading;
+
+namespace Iot.Device.Imu
+{
+    /// <summary>
+    ///  MPU9250 class. MPU6500 has an embedded gyroscope, accelerometer and temperature.
+    /// </summary>
+    public class Mpu6500 : IDisposable
+    {
+        private const float Adc = 0x8000;
+        private const float Gravity = 9.807f;
+        internal I2cDevice _i2cDevice;
+        private Vector3 _accelerometerBias = new Vector3();
+        private Vector3 _gyroscopeBias = new Vector3();
+        private AccelerometerRange _accelerometerRange;
+        private GyroscopeRange _gyroscopeRange;
+        private AccelerometerBandwidth _accelerometerBandwidth;
+        private GyroscopeBandwidth _gyroscopeBandwidth;
+        internal bool _wakeOnMotion;
+
+        /// <summary>
+        /// Default address for MPU9250
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x68;
```

Comment:
constants should be listed at the top of the class (move them up) with public being first